hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockRecoveryWorker.java

```diff
@@ -436,9 +442,17 @@ protected void recover() throws IOException {
                   "datanode={})", block, internalBlk, id, e);
         }
       }
-      checkLocations(syncBlocks.size());
 
-      final long safeLength = getSafeLength(syncBlocks);
+      final long safeLength;
+      if (dnNotHaveReplicaCnt + zeroLenReplicaCnt <= locs.length - ecPolicy.getNumDataUnits()) {
+        checkLocations(syncBlocks.size());
+        safeLength = getSafeLength(syncBlocks);
+      } else {
+        safeLength = 0;
+        LOG.warn("Block recovery: More than {} datanodes do not have the replica of block {}." +
```

Comment:
Suggest printing out the value of `zeroLenReplicaCnt` as well.

Reply:
Fixed it. Thanks sir.

Reply:
What does this "More than" mean?

Reply:
It seems useless here, have removed it . Originally, it was more than (9 - 6) datanodes ...